api/envoy/extensions/filters/http/grpc_http1_bridge/v3/config.proto

```diff
@@ -18,4 +18,7 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 message Config {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.grpc_http1_bridge.v2.Config";
+
+  // If true, the filter sets the request's HTTP status code alongside the existing gRPC status code.
```

Comment:
I'm a bit confused on this one; it looks like HTTP status code was being set before for regular gRPC response. It sounds like this PR is a bug fix and expanding this behavior to trailers-only response. What is the meaning/purpose of this field if this is true?

Reply:
This PR was [merged previously and reverted](https://github.com/envoyproxy/envoy/pull/15317) after realizing that deployments relied on failed gRPC requests to return a 200 HTTP status. With the revert, there was an ask to wrap the "bug fix" with a configuration and allow users to opt-in if they prefer to. See linked PR for details. Personally, I agree with the request from an observability point of view since it could cause confusion for some gRPC requests to have a "useful" HTTP status code while others don't. 

Also just wanted to clarify, `it looks like HTTP status code was being set before for regular gRPC response.` Correct me if I'm wrong, but I believe the HTTP status code is only set for gRPC requests that go through one of the special grpc filters (bridge/transcoder). Otherwise, all other requests should default to 200.

Reply:
What I'm confused about is the API talks about HTTP status codes in general, but the PR title and implementation only apply this to trailers-only responses.

Reply:
Apologies for the delay, I wanted to prioritize another OSS issue before coming back to this. I can rewrite the description to be ` // If true, the filter sets the request's HTTP status code alongside the existing gRPC status code in a trailers-only response.` to provide more clarity.

Reply:
Maybe something like 
```
// Controls whether the HTTP status code transformation logic should apply to headers-only responses. The HTTP status code will be set according to https://cloud.google.com/apis/design/errors#handling_errors whenever the grpc-status is not OK. By default this logic does not apply to headers-only responses.
```
?

It's a bit unfortunate that we can't just treat this is a bug fix, but I think this has been out in the wild long enough that there is going to be code that will blow up if Envoy starts responding non-200, so I'm not sure its worth the risk to try to flip this for everyone, even with a runtime flag.

Reply:
Yes, something like what @snowp wrote (trailers-only rather than headers-only?).